x11-libs/lexilla/lexilla-5.2.3.recipe

```diff
@@ -0,0 +1,104 @@
+SUMMARY="A free library of language lexers for use with Scintilla"
+DESCRIPTION="Lexilla is a free library of language lexers that can be used \
+with the Scintilla editing component. It comes with complete source code and \
+a license that permits use in any free project or commercial product.
+Originally, this functionality was incorporated inside Scintilla. \
+It has been extracted as a separate project to make it easier for \
+contributors to work on support for new languages and to fix bugs in existing \
+lexers. It also defines a protocol where projects can implement their own \
+lexers and distribute them as they wish."
+HOMEPAGE="https://scintilla.org"
+COPYRIGHT="1998-2022 Neil Hodgson"
+LICENSE="Scintilla"
+REVISION="1"
+SOURCE_URI="https://prdownloads.sf.net/scintilla/lexilla${portVersion//\./}.tgz"
+CHECKSUM_SHA256="1694b3d82037458fb12064530e6ab3f6a909085d09bb8933dde823135fe56454"
+SOURCE_DIR="lexilla"
+srcGitRev2="e391de7b72f7cc58c8d267524d9581e1a875d6e9"
+SOURCE_URI_2="https://github.com/KapiX/scintilla-haiku-lexers/archive/$srcGitRev2.tar.gz"
+CHECKSUM_SHA256_2="a3a0121dd641355fbb1a5ff8fe8896d574ad809975867a5ffdefc609448194c7"
+SOURCE_DIR_2="scintilla-haiku-lexers-$srcGitRev2"
+SCINTILLA_VERSION="5.3.0"
+SOURCE_URI_3="https://prdownloads.sf.net/scintilla/scintilla${SCINTILLA_VERSION//\./}.tgz"
+CHECKSUM_SHA256_3="fd5260f56de44cc9d5e2d90086ec6ffe7096abbdc91bc9c1c45f5289cd40407b"
+SOURCE_DIR_3="scintilla"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	lexilla$secondaryArchSuffix = $portVersion
+	lib:liblexilla$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	lexilla${secondaryArchSuffix}_devel = $portVersion
+	devel:liblexilla$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	lexilla$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	makefile_engine
+	cmd:ar
+	cmd:g++$secondaryArchSuffix
+	cmd:make
+	devel:libscintilla$secondaryArchSuffix >= ${SCINTILLA_VERSION%%.*}
+	"
+
+#PATCH()
+#{
+#	#sed -i -e 's#SymbolValue() noexcept = #SymbolValue() = #' \
+#	#	$sourceDir/lexers/LexCPP.cxx
+#
+#	local packageLinksDir=$(dirname $portPackageLinksDir)
+#	local includeDir="$packageLinksDir/$portName-$portFullVersion/devel~libscintilla$secondaryArchSuffix/$relativeIncludeDir/scintilla"
+#	#sed -i -e "s#../../scintilla/include#$includeDir#g" \
+#	#	$sourceDir/src/makefile
+#}
+
+BUILD()
+{
+	local SCINTILLA_INCLUDE=$sourceDir/../scintilla/include
+	mkdir -p $SCINTILLA_INCLUDE
+	cp -R $sourceDir3/include/* $SCINTILLA_INCLUDE
+
+	#local packageLinksDir=$(dirname $portPackageLinksDir)
+	#local includeDir="$packageLinksDir/$portName-$portFullVersion/devel~libscintilla$secondaryArchSuffix/$relativeIncludeDir/scintilla"
+	#ls -la $packageLinksDir/$portName-$portFullVersion/
```

Comment:
I'd prefer to use packageLinks instead of downloading scintilla source, but for whatever reason devel~libscintilla points to `'?'`. Any tips?

Reply:
No, but I suppose `$portPackageLinksDir/devel~libscintilla$secondaryArchSuffix/` should work.

Reply:
I couldn't make it work. It might be a bug in haikuporter.